trestle/common/file_utils.py

```diff
@@ -264,3 +264,29 @@ def relative_resolve(candidate: pathlib.Path, cwd: pathlib.Path) -> pathlib.Path
     except ValueError:
         raise TrestleError(f'Provided dir {candidate} is not relative to {cwd}')
     return new
+
+
+def insert_text_in_file(file_path: pathlib.Path, tag: Optional[str], text: str) -> bool:
+    r"""Insert text lines after line containing tag.
+
+    Return True on success, False tag not found.
+    Text is a string with appropriate \n line endings.
+    If tag is none just add at end of file.
+    """
+    if not file_path.exists():
+        raise TrestleError(f'Test file {file_path} not found.')
+    if tag:
+        lines: List[str] = []
+        with file_path.open('r') as f:
+            lines = f.readlines()
+        for ii, line in enumerate(lines):
+            if line.find(tag) >= 0:
+                lines.insert(ii + 1, text)
+                with file_path.open('w') as f:
+                    f.writelines(lines)
+                return True
```

Comment:
Just because I am reading the whole file in memory, then writing it out if the tag is found.  If the tag is not found I don't need to write.  I'm a little concerned about reading and writing at the same time.

Reply:
Might have missed the point that you do two different operations here, it is fine as is